code/obj/item/grenades.dm

```diff
@@ -934,7 +935,10 @@ ADMIN_INTERACT_PROCS(/obj/item/gimmickbomb, proc/arm, proc/detonate)
 	proc/beep(i)
 		var/k = i/2
 		sleep(k*k)
-		flick(icon_state+"_beep", src)
+		if (src.icon_state_armed)
+			src.icon_state = icon_state_armed
+		if (istype(src, /obj/item/gimmickbomb/gold))
+			src.setMaterial(getMaterial("gold"))
```

Comment:
this should be handled in the subtype not in the abstract parent

Reply:
Transferred it to the Subtype by including the gold material setting in a modification of the beep proc. Seems to work right.